app/src/main/java/io/apicurio/registry/mt/TenantReaper.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.mt;
+
+import io.apicurio.multitenant.api.beans.TenantStatusValue;
+import io.apicurio.multitenant.api.datamodel.RegistryTenant;
+import io.apicurio.registry.storage.RegistryStorage;
+import io.apicurio.registry.types.Current;
+import io.quarkus.scheduler.Scheduled;
+import org.slf4j.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Random;
+
+/**
+ * Periodically cleanup data of tenants marked as deleted.
+ *
+ * @author Jakub Senko <jsenko@redhat.com>
+ */
+@ApplicationScoped
+public class TenantReaper {
+
+    @Inject
+    Logger log;
+
+    @Inject
+    MultitenancyProperties properties;
+
+    @Inject
+    TenantMetadataService tenantService;
+
+    @Inject
+    @Current
+    RegistryStorage storage;
+
+    @Inject
+    TenantContextLoader tcl;
+
+    @Inject
+    TenantContext tctx;
+
+    Instant next;
+
+    @PostConstruct
+    void init() {
+        if (!properties.isMultitenancyEnabled())
```

Comment:
could you add curly brackets here?

Reply:
Sure, I'm used to avoid them for one-liner conditions.